File: Algorithms/Maximum subarray sum/Algorithm2.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
  int n;
  cout<<"Enter the length of the array\n";
  cin>>n;
  int A[n];
  for(int i=0;i<n;i++)
    cin>>A[i];
  //Calculating every subarray and simply finding best
  int best=0;
  int tempsum=0;
  for(int i=0;i<n;i++)
  {
    tempsum=0;//reset here
    for(int j=i;j<n;j++)
    {
      tempsum+=A[j];//this inevitably sums each subarray separately
      best=max(tempsum,best);
    }
  }
  cout<<"\nMax sum=" <<best;
  //Time complexity has been reduced to O(n^2)
}

File: Algorithms/Maximum subarray sum/Algorithm2.py
A=[]
while 1:
    x=input("Enter array(exit to end)")
    if x=='exit':
        break
    A.append(int(x))
best=0
tempsum=0
for x in range(len(A)):
    tempsum=0
    for y in A[x:]:
        tempsum+=y
        if best<tempsum:
            best=tempsum
print("Max sum="+str(best))
#complexity is O(n^2)
